### PR TITLE
Remove MSALErrorKeychainFailure error

### PR DESCRIPTION
**Pull request by unpluggedk**
#127 
Remove MSALErrorKeychainFailure error and return error code directly with NSOSStatusErrorDomain

### CODE REVIEW

File: MSAL/src/MSALError_Internal.h [hunk 1]
```diff
@@ -79,5 +79,7 @@ extern void MSALFillAndLogError(NSError * __autoreleasing *, id<MSALRequestConte
 
 #define REQUIRED_PARAMETER_ERROR(_PARAMETER, _CTX) MSALFillAndLogError(error, _CTX, MSALErrorInvalidParameter, nil, nil, nil, __FUNCTION__, __LINE__, @#_PARAMETER " is a required parameter and must not be nil or empty.")
 
-// Convenience macro for creating a cache error
-#define MSAL_KEYCHAIN_ERROR_PARAM(_CTX, _OSERROR, _OPERATION) MSAL_OSERROR_PARAM(_CTX, MSALErrorKeychainFailure, _OSERROR, @"Keychain failure (%d) while %@", (int)_OSERROR, _OPERATION)
+// Convenience macro for creating a keychain error
+#define MSAL_KEYCHAIN_ERROR(_CTX, _OSSTATUS) \
```
Comment (RPangrle) [inline MSAL/src/MSALError_Internal.h]: You absolutely need to keep the operation and the coherent keychain logging here so we have some idea what it was doing when it failed. It's also useful to include that string in the user dictionary under the MSALErrorDescriptionKey

Comment (unpluggedk) [inline MSAL/src/MSALError_Internal.h]: 👍

**Comment by unpluggedk** [comment]
Revising the existing macros.

**Comment by unpluggedk** [comment]
Revised error-internal.
Ready for review.

File: MSAL/test/unit/MSALTokenCacheTests.m [hunk 2]
```diff
@@ -881,7 +881,7 @@ - (void)testUserForIdentifier_whenUserNotInCacheAndNoError_shouldFailAndStillLog
                                      clientId:@"12345"
                                   environment:@"environment.com"
                                         error:nil]);
-    
+    NSLog(@"%@", [[MSALTestLogger sharedLogger] lastMessage]);
```
Comment (oldalton) [inline MSAL/test/unit/MSALTokenCacheTests.m]: Why is this necessary?

Comment (unpluggedk) [inline MSAL/test/unit/MSALTokenCacheTests.m]: Is not. Deleted.

File: MSAL/test/unit/MSALPublicClientApplicationTests.m [hunk 3]
```diff
@@ -844,12 +844,13 @@ - (void)testUserKeychainError
...
          (void)user;
          (void)clientId;
-         MSAL_KEYCHAIN_ERROR_PARAM(ctx, -34018, @"fetching team ID.");
+         MSAL_KEYCHAIN_ERROR(ctx, -34018, @"fetching team ID.");
          return NO;
      }];
     
     XCTAssertFalse([application removeUser:user error:&error]);
     XCTAssertNotNil(error);
+    XCTAssertTrue(error.domain == NSOSStatusErrorDomain);
```
Comment (oldalton) [inline MSAL/test/unit/MSALPublicClientApplicationTests.m]: Maybe XCTAssertEqualObjects would be better here?
